dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java

```diff
@@ -0,0 +1,123 @@
+	private final PermissionAPI   permissionAPI;
+
+	public SaveContentAsDraftActionlet() {
+
+		this.contentletAPI   = APILocator.getContentletAPI();
+		this.relationshipAPI = APILocator.getRelationshipAPI();
+		this.categoryAPI     = APILocator.getCategoryAPI();
+		this.permissionAPI   = APILocator.getPermissionAPI();
+	}
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
+	public String getName() {
+		return "Save Draft content";
+	}
+
+	public String getHowTo() {
+
+		return "This actionlet will do checkin as draft the content.";
+	}
+
+	public void executeAction(final WorkflowProcessor processor,
+							  final Map<String, WorkflowActionClassParameter> params) throws WorkflowActionFailureException {
+
+		final Map<Relationship, List<Contentlet>> contentRelationships = new HashMap<Relationship, List<Contentlet>>();
+
+		try {
+
+			final Contentlet contentlet             = processor.getContentlet();
+			final User user                         = processor.getUser();
+			final List<Category > categories        =
+					this.categoryAPI.getParents(contentlet, user, false);
+			final List< Permission > permissions    =
+					this.permissionAPI.getPermissions(contentlet, false, true);
+
+			Logger.debug(this,
+					"Saving the content as draft of the contentlet: " + contentlet.getIdentifier());
+
+			contentlet.setProperty(Contentlet.WORKFLOW_IN_PROGRESS, Boolean.TRUE);
+
+			this.buildRelationships(processor.getUser(), contentRelationships, contentlet);
+
+			this.contentletAPI.saveDraft(
+					processor.getContentlet(), contentRelationships, categories, permissions, user, true);
+
+			Logger.debug(this,
+					"content draft already saved for the contentlet: " + contentlet.getIdentifier());
+		} catch (Exception e) {
+
+			Logger.error(this.getClass(),e.getMessage(),e);
+			throw new  WorkflowActionFailureException(e.getMessage());
+		}
+	}
+
+	private void buildRelationships(final User user,
+									final Map<Relationship, List<Contentlet>> contentRelationships,
+									final Contentlet contentlet) throws DotDataException, DotSecurityException {
+
+		final List<Relationship> relationships =  this.relationshipAPI.
+				byContentType(contentlet.getContentType());
+
+		for (final Relationship relationship : relationships) {
+
+            if (!contentRelationships.containsKey(relationship)) {
+
+                contentRelationships
+                        .put(relationship, new ArrayList<Contentlet>());
+            }
+
+            final List<Contentlet> relatedContents = this.contentletAPI.getRelatedContent(
+                    contentlet, relationship, user, true);
+
+            for (final Contentlet relatedContent : relatedContents) {
+                contentRelationships.get(relationship).add(relatedContent);
+            }
+        }
+	}
+
+	public WorkflowStep getNextStep() {
+
+		return null;
+	}
+
+	@Override
+	public  List<WorkflowActionletParameter> getParameters() {
+
+		return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)